ui/src/app/pages/homepage/homepage.tsx

```diff
@@ -229,10 +230,9 @@ export const Homepage = withUserProfile()(class extends React.Component<Props, S
   }
 
   async checkWorkspaces() {
-    workspacesApi().getWorkspaces().then(response => {
-      this.setState({
-        userHasWorkspaces: response.items.length > 0
-      });
+    const userWorkspaces = (await fetchWithGlobalErrorHandler(() => workspacesApi().getWorkspaces())).items;
```

Comment:
is there any chance userWorkspace can be null or undefined? should userHasWorkspaces be like userWorkspace && userWorkspaces.length >0 ?